### ISSUE TITLE
XQuartz hanging with Ventura 13.4

### ISSUE THREAD

**Issue by mdcowles** [XQuartz/XQuartz#347]
I have a Python Tkinter app which has run for years with XQuartz (I compile Python and Tcl and Tk to use X11). With today's Ventura update, it runs but hangs after opening and closing 3 or 4 windows and XQuartz must be force-quit. I've sent a couple of dumps to Apple. After force-quitting, XQuartz can be run again, but my app won't start: the Tcl library complains couldn't connect to display ":0.0"

Let me know if there's anything else I can tell you.

**Comment by jeremyhu** [comment]
Can you provide the feedback report numbers, so I can check on them?
Please be sure to provide a sysdiagnose focused on your hung process.

Also, you should not be using `DISPLAY=:0.0`.  This indicates that you are hard-coding DISPLAY somewhere, overriding the value we set (which is to a path to a launchd socket).  You probably got lucky that XQuartz was using :0 and on re-launch, it moved to :1 because :0 was not torn down correctly.

**Comment by mdcowles** [comment]
I didn't keep track of the incident identifiers I sent earlier, but I just sent two more:

6FC0791A-4DC9-4526-B1C7-09D7CCF20AC7

EAB29C8E-024C-48C9-ACE2-570A1808AE52

What shall I do with the output from sysdiagnose? The machine is my personal laptop. I doubt that it captured anything especially private, but unless you say I should, I'm a trifle reluctant to post it here.

If I don't set $DISPLAY, tcl complains:

no display name and no $DISPLAY environment variable

If there is something better to set it to than ":0.0", I'd be glad to know.

It may not be relevant, but it seems to me that 13.4 is a bit crashy. My MacBook wouldn't come out of sleep yesterday morning and I had what seemed to be a crash of the whole login session this afternoon.

**Comment by jeremyhu** [comment]
I don't know what thost UUIDs correspond to.

I'm looking for the feedback report number (eg: FB123455) which you get after reporting your feedback.  You can also see it if you check in on your bug report at http://feedback.apple.com

I suggest you attach the sysdiagnose to your feedback report.  Once you provide the FB number, I can get access to it.

--

The $DISPLAY environment variable should be getting set by launchd.  Check the output of `launchctl getenv DISPLAY`.

Note that in the next version of XQuartz, I think setting DISPLAY to :0 will be illegal as part of addressing CVE-2020-25697.  See https://lists.x.org/archives/xorg-announce/2023-June/003399.html

**Comment by mdcowles** [comment]
I posted FB12229013 which includes the output from sysdiagnose.

Those other IDs were the IDs that the macOS gave me for the reports it said it was sending to Apple after I force-quit XQuartz. I tried a couple of things today and got the same hangs, but the OS didn't offer to send more reports.

You're quite right that setting $DISPLAY was unnecessary. I don't remember now why it seemed to be necessary back when I did. Leaving it as the path to the socket produced the same results.

**Comment by mdcowles** [comment]
Upgrading to macOS 13.4.1 seems not to have made a difference.

**Comment by jeremyhu** [comment]
The FB report for this became rdar://110284144

**Comment by jeremyhu** [comment]
@javacookie123 Given the DISPLAY variable you report, are you sure you're not hitting the SSH tunnel timeout?

https://superuser.com/questions/407218/why-does-ssh-x-forwarding-timeout-after-a-while

That is unrelated to the issue being discussed here.  If you still have issues, please open a new ticket.

**Comment by jeremyhu** [comment]
Ugh.  This one was my fault... trying to fix a few other crashes, I introduced deadlock.  Thanks for the report.  Working with the team that owns this now to come up with a solution.

**Comment by mdcowles** [comment]
My feedback post at:

https://feedbackassistant.apple.com/feedback/12229013

has gotten what appears to be an automatic update from Apple that begins:

    Please verify this issue with macOS 14 beta 4 and update your bug report
    with your results by logging into https://feedbackassistant.apple.com/ or
    by using the Feedback Assistant app.

It also asks for new sysdiagnose output with the new version of the OS.

I assume that that's unnecessary because you have found the bug and are working on it. I hope the fact that I don't normally run the betas won't cause a delay.

**Comment by sorodonhobbs** [comment]
Any schedule on when a fix might be available?  Or is there advice on any workaround?.  Maybe I could avoid using local xterm-s but could still allow remote windows to display on my Mac?

I was just required to upgrade to Mac OS 13.4.1 (Ventura) and then XQuartz started freezing about 5 minutes after each time I restarted it.  I upgraded Xquartz to 2.8.6_beta2 (xorg-server 21.1.8) hoping XQuartz 2.8.6 would now include a fix.  However, the fix does not seem to be there since it is still freezing.

**Comment by jeremyhu** [comment]
Not really.  There isn't really anybody that really works on X11 any more... so it's not likely to get fixed.

**Comment by jeremyhu** [comment]
This was fixed in macOS 14
